### ISSUE TITLE
On-demand Revalidation in all dynamic paths from specific route

### ISSUE THREAD

**Issue by joao-pedrozo** [vercel/next.js#38266]
### Describe the feature you'd like to request

Today is possible to revalidate ONE route with res.revalidate()
```
// pages/api/revalidate.js

export default async function handler(req, res) {
    await res.revalidate('/path-to-revalidate')
    return res.json({ revalidated: true })
}
```

But it would be interesting if we could revalidate all paths from a dynamic route, like above

```
// pages/api/revalidate.js

export default async function handler(req, res) {
    await res.revalidate('/posts/*')
    return res.json({ revalidated: true })
}
```

### Describe the solution you'd like

Accept a regex or a pattern in res.revalidate() so we could revalidate multiple paths at once.

### Describe alternatives you've considered

Accept regex or a pattern in params to trigger the bulk revalidation.

**Comment by 11koukou** [comment]
Yes that would be useful indeed. However it seems demanding in terms of performance, not sure though
